provision/Provision-All.ps1

```diff
@@ -1,66 +1,83 @@
-
+[CmdletBinding()]
 Param
 (
-	[Parameter (Mandatory = $true)]
-    [ValidatePattern("^[A-Za-z][-A-Za-z0-9]*[A-Za-z0-9]$")]               # needs to start with letter or number, and contain only letters, numbers, and hyphens.
-    [String]$ServiceBusNamespace ="fabrikam-ns01",                                   
-
-    # dont use this: [ValidatePattern("^[a-z0-9]*$")]  # don't use this, powershell script is case insensitive, uppercase letter still pass as valid 
-    [ValidateScript({
-      # we need to use cmathch which is case sensitive, don't use match
-      If ($_ -cmatch "^[a-z0-9]*$") {                         # needs contain only lower case letters and numbers.
-        $True
-      }else {
-        Throw "`n---Storage account name can only contain lowercase letters and numbers!---"
-      }
-    })]
-	[Parameter (Mandatory = $true)]
-    [String]$StorageAccountName = "fabrikamstorage01",    
+	[ValidateNotNullOrEmpty()]
+	[Parameter(Mandatory = $True)]
+	[string]$SubscriptionName,
 
-	[Parameter (Mandatory = $true)]
-    [String]$StreamAnalyticsJobName = "fabrikamstreamjob01",
+	[ValidateNotNullOrEmpty()]
+	[Parameter (Mandatory = $True)]
+	[String]$ServiceBusNamespace,                                   
 
-	[Parameter (Mandatory = $true)]
-	[string]$SubscriptionName = "Azure Guidance",
+	[ValidateNotNullOrEmpty()]
+	[Parameter (Mandatory = $True)]
+	[String]$StorageAccountName,    
```

Comment:
you need keep the validations:
[ValidateScript({  
11   -      If ($_ -cmatch "^[a-z0-9]*$") {                         # needs contain only lower case letters and numbers.  
12   -        $True  
13   -      }else {  
14   -        Throw "`n---Storage account name can only contain lowercase letters and numbers!---"  
15   -      }  
16   -    })]  
17   -	[Parameter (Mandatory = $true)]  
18   -    [String]$StorageAccountName = "fabrikamstorage01",